backend/apps/matching/views/staff_approvals.py

```diff
@@ -53,18 +62,27 @@ def get(self, request, p_type):
 
     def post(self, request, *args, **kwargs):
         post_params = self.request.POST
+        p_type = self.kwargs["p_type"]
         if "delete" in post_params:
-            p = get_object_or_404(Participant[self.kwargs["p_type"]], uuid=post_params["uuid"])
+            if not request.user.has_perm("matching.delete_participant%s" % p_type.lower()):
```

Comment:
I would also test for `matching.delete_participant%s` and `matching.can_approve_type_%s` with a class-based decorator. A person who can't change the values, should probably also not see these pages.

Reply:
I don't understand what you mean - those get tested in lines 27-30. These here are for the actual action. 

I think using a class decorator would be less expressive as it would probably have a long name. If you use them separately as decorators, both would be required. But I think not every approver would have the rights to delete participants and vice versa?

Reply:
You're right, I missed that. It would probably be more consistent to have a method decorator and not have decorators on classes, decorators on methods and checks in a method (which makes it hard to check with a quick glance if the right permissions are set). But I see that the decorator is also not totally self-explanatory.